src/server/stat_job.c

```diff
@@ -464,16 +455,14 @@ status_subjob(job *pjob, struct batch_request *preq, svrattrl *pal, int subj, pb
 	/* when eligible_time_enable is off,				      */
 	/* clear the set flag so that eligible_time and accrue_type dont show */
 	if (server.sv_attr[(int)SVR_ATR_EligibleTimeEnable].at_val.at_long == 0) {
-		oldeligflags = pjob->ji_wattr[(int)JOB_ATR_eligible_time].at_flags;
-		mark_jattr_not_set(pjob, JOB_ATR_eligible_time);
-		pjob->ji_wattr[(int)JOB_ATR_eligible_time].at_flags |= ATR_MOD_MCACHE;
+		attribute *attr = get_jattr(pjob, JOB_ATR_eligible_time);
 
-		oldatypflags = pjob->ji_wattr[(int)JOB_ATR_accrue_type].at_flags;
-		mark_jattr_not_set(pjob, JOB_ATR_accrue_type);
-		pjob->ji_wattr[(int)JOB_ATR_accrue_type].at_flags |= ATR_MOD_MCACHE;
+		oldeligflags = attr->at_flags;
+		ATR_UNSET(attr);
 
-		/* Note: ATR_VFLAG_MODCACHE must be set because of svr_cached() does */
-		/* 	 not correctly check ATR_VFLAG_SET */
+		attr = get_jattr(pjob, JOB_ATR_accrue_type);
+		oldatypflags = attr->at_flags;
+		ATR_UNSET(attr);
```

Comment:
same comment here about using job level function mark_jattr_not_set() instead of ATR_UNSET directly